qiskit/transpiler/passes/utils/check_map.py

```diff
@@ -20,18 +20,37 @@ class CheckMap(AnalysisPass):
     """Check if a DAG circuit is already mapped to a coupling map.
 
     Check if a DAGCircuit is mapped to `coupling_map` by checking that all
-    2-qubit interactions are laid out to be physically close, setting the
-    property ``is_swap_mapped`` to ``True`` or ``False`` accordingly.
+    2-qubit interactions are laid out to be on adjacent qubits in the global coupling
+    map of the device, setting the property ``is_swap_mapped`` to ``True`` or ``False``
+    accordingly. Note this does not validate directionality of the connectivity between
+    qubits. If you need to check gates are implemented in a native direction
+    for a target use the :class:`~.CheckGateDirection` pass instead.
     """
 
-    def __init__(self, coupling_map):
+    def __init__(self, coupling_map=None, target=None):
         """CheckMap initializer.
 
         Args:
             coupling_map (CouplingMap): Directed graph representing a coupling map.
+            target (Target): A target representing the target backend, if both
+                ``coupling_map`` and this are specified then this argument will take
+                precedence and ``coupling_map`` will be ignored.
         """
         super().__init__()
-        self.coupling_map = coupling_map
+        if coupling_map is None and target is None:
+            self.qargs = None
+        else:
+            self.qargs = set()
+            if target is not None:
+                if target.qargs is not None:
+                    for edge in target.qargs:
+                        if len(edge) == 2:
+                            self.qargs.add(edge)
+                            self.qargs.add((edge[1], edge[0]))
+            else:
+                for edge in coupling_map.get_edges():
+                    self.qargs.add(edge)
+                    self.qargs.add((edge[1], edge[0]))
```

Comment:
Pretty minor, but if we're using the generic coupling map implementation, maybe it's better to do `Target.build_coupling_map` and then always get the edges from that so we're definitely doing things in a unified way across all passes?

I don't know if there was a reason you deliberately didn't do that, though.

Reply:
I think that I did this just to avoid the overhead for coupling map construction. But at this point we'll have already created it once so it'll be cached in the target so we don't have any extra overhead here for the construction anyway. I'll update this to unify the code path.